sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -0,0 +1,221 @@
+        }
+
+        return new DigitalTwinsAsyncClient(this.httpPipeline, this.serviceVersion, this.endpoint);
+    }
+
+    /**
+     * Set the service endpoint that the built client will communicate with. This field is mandatory to set.
+     *
+     * @param endpoint URL of the service.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Set the authentication token provider that the built client will use for all service requests. This field is
+     * mandatory to set unless you set the http pipeline directly and that set pipeline has an authentication policy configured.
+     *
+     * @param tokenCredential the authentication token provider.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder tokenCredential(TokenCredential tokenCredential) {
+        this.tokenCredential = tokenCredential;
+        return this;
+    }
+
+    /**
+     * Sets the {@link DigitalTwinsServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param serviceVersion The service API version to use.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder serviceVersion(DigitalTwinsServiceVersion serviceVersion) {
+        this.serviceVersion = serviceVersion;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpClient} to use for sending a receiving requests to and from the service.
+     *
+     * @param httpClient HttpClient to use for requests.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpLogOptions} for service requests.
+     *
+     * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     * @throws NullPointerException If {@code logOptions} is {@code null}.
+     */
+    public DigitalTwinsClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.logOptions = Objects.requireNonNull(logOptions, "'logOptions' cannot be null.");
```

Comment:
`HttpLoggingPolicy` allows for `null` `HttpLogOptions` to effectively be a no-op, don't think it needs to be `null` checked here.